redhat-appstudio/remote-secret
# Implement copying of RS data using dataFrom field, including checking for appropriate user permissions in the webhook

### What does this PR do?

This reworks the webhook code so that we can take advantage of the lower level implementation that has access to the `UserInfo` which contains the user that is invoking the request.

This then enables us to check for the permissions of such user before allowing them to copy the data from one remote secret to another.

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/SVPI-553


### How to test this PR?
1. Apply the `samples/remotesecret-with-copied-data.yaml`
2. This should create 2 remote secrets in the default namespace, both in the injected state.
3. The first one, `myoriginalrs`, is created first, with the `data` field so that it directly goes into the injected state.
4. The second one, `myrs1`, is created by copying the data of the first one using the new `dataFrom` field and should therefore also directly enter the injected state.

Image to try the functionality:
`quay.io/lkrejci/remote-secret-controller:svpi-606`
